lib/CL/poclcc_binary.c

```diff
@@ -0,0 +1,443 @@
+#include "poclcc_binary.h"
+
+void poclcc_free(poclcc_global *binary_format){
```

Comment:
What we dump in the binary file is the _poclcc_device_ struct.
Maybe we don't what this struct to be named _poclcc_global_ but I'm not sure that _pocl_binary_ matchs it well.
Maybe something like _poclcc_binaries_ but it could still be confusing...